test/test_bmi_model.py

```diff
@@ -41,14 +43,90 @@ def test_bmi_initialize(config: Config) -> None:
     Args:
         config (Config): configuration to initialize the model with
     """
-    model: Bmi_Model = Bmi_Model()
-    assert not hasattr(model, "model")
-
-    model.initialize(config)
-
-    assert hasattr(model, "model")
-    assert model.model != None
-
-    assert model.learning_rate == config.learning_rate
-    assert model.optimizer != None
-
+    assert not hasattr(bmi_model, "model")
+
+    bmi_model.initialize(config)
+
+    assert hasattr(bmi_model, "model")
+    assert bmi_model.model != None
+
+    assert bmi_model.learning_rate == config.learning_rate
+    assert bmi_model.optimizer != None
+
+@pytest.mark.parametrize('model', ['bmi_model', 'bmi_model_initialized'])
+def test_bmi_component_name(model, request):
+    m = request.getfixturevalue(model)
+    assert m.get_component_name() == "Tensor Test"
+
+@pytest.mark.parametrize('model', ['bmi_model', 'bmi_model_initialized'])
+def test_bmi_input_item_count(model, request):
+    m = request.getfixturevalue(model)
+    assert m.get_input_item_count() == 1
+
+@pytest.mark.parametrize('model', ['bmi_model', 'bmi_model_initialized'])
+def test_bmi_input_var_names(model, request):
+    m = request.getfixturevalue(model)
+    assert m.get_input_item_count() == 1
+    names = m.get_input_var_names()
+    assert len(names) == 1
+    assert names[0] == "precipitation"
+
+@pytest.mark.parametrize('model', ['bmi_model', 'bmi_model_initialized'])
+def test_bmi_output_item_count(model, request):
+    m = request.getfixturevalue(model)
+    assert m.get_output_item_count() == 1
+
+@pytest.mark.parametrize('model', ['bmi_model', 'bmi_model_initialized'])
+def test_bmi_output_var_names(model, request):
+    m = request.getfixturevalue(model)
+    assert m.get_output_item_count() == 1
+    names = m.get_output_var_names()
+    assert len(names) == 1
+    assert names[0] == "runoff"
+
+@pytest.mark.parametrize("var,expected", [
+                            ("precipitation", 0),
+                            ("runoff", 0)
+                        ])
+def test_bmi_var_grid(bmi_model_initialized, var, expected):
+    m = bmi_model_initialized
+    assert m.get_var_grid(var) == expected
+
+@pytest.mark.parametrize("var", ["var1", "var2"])
+def test_bmi_var_grid_2(bmi_model_initialized, var):
+    m = bmi_model_initialized
+    with pytest.raises(UnknownBMIVariable):
+            m.get_var_grid(var)
+
+@pytest.mark.parametrize("input", [
```

Comment:
these should be `torch.tensor()` rather than `torch.Tensor()` (see above for reason)